deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -324,6 +324,19 @@ def get(self, request, name, stage=None):
         env_tag = tags_helper.get_latest_by_targe_id(request, env['id'])
         basic_cluster_info = None
         capacity_info = {'groups': groups}
+
+        stage_with_external_id= None
+        for env_stage in envs:
+            if env_stage['externalId'] is not None:
+                stage_with_external_id = env_stage
+                break
```

Comment:
It is possible that one of the stage in the environment  has a nimbus id that is not default project but the other stages in the environment still are default project. I think it is better to find out the stage this page is displaying and use that stage.

Reply:
Thanks for pointing out, updated.